kura/org.eclipse.kura.core/src/main/java/org/eclipse/kura/core/data/DataServiceImpl.java

```diff
@@ -750,7 +750,11 @@ private void createAutoConnectStrategy() {
             return;
         }
 
-        final Optional<CronExpression> schedule = this.dataServiceOptions.getConnectionScheduleExpression();
+        Optional<CronExpression> schedule = Optional.empty();
+        
+        if (this.dataServiceOptions.isConnectionScheduleEnabled()) {
```

Comment:
I would not do the check here but rather in the options class since it is already returning an Optional: https://github.com/eclipse/kura/blob/31ef269ef081bedd72772b8508dcc98cc960ba37/kura/org.eclipse.kura.core/src/main/java/org/eclipse/kura/core/data/DataServiceOptions.java#L183